cou/steps/backup.py

```diff
@@ -76,12 +82,12 @@ def get_database_app(model_name: Optional[str] = None) -> Optional[str]:
     :param model_name: Name of model to query.
     :type model_name: str
     :returns: Name of the mysql-innodb-cluster application name
-    :rtype: str
+    :rtype: ApplicationStatus
```

Comment:
iiuc this should be a `juju.application.Application`

Reply:
ok

Reply:
nvm, it really is an ApplicationStatus
```
In [8]: type(await get_database_app())
Out[8]: juju.client._definitions.ApplicationStatus
```
So I'd suggest making this function more consistent: it's called `get_database_app`, its stated purpose is `"""Get mysql-innodb-cluster application name.`, but it actually returns a `juju.client._definitions.ApplicationStatus` or raises a `UnitNotFound`.

If you're only going to use it internally to then be able to address the application by name I'd go with something like `get_db_app_name` and a rtype of String (and a `UnitNameNotFound` exception); aligning on Application or ApplicationStatus would be fine too though

Reply:
I did not want to introduce new exception so I've renamed function to `get_database_app_unit_name` returns the name of the unit of database application and raises `UnitNotFound` if it is unable to find one. It makes more sense now.